### PR TITLE
Adding support for the Maxwell CDC format

### LINKED ISSUE #335: Document support for the Maxwell CDC format
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/6057

### Which part(s) of the docs might be affected or should be updated? And how?

https://www.risingwave.dev/docs/latest/sql-create-source/
https://www.risingwave.dev/docs/latest/create-source-cdc/

Previously, only Debezium_JSON is supported as a CDC format, but this PR adds the support for Maxwell format. 
The syntax is `ROW FORMAT MAXWELL`. 

### Reference

https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/table/formats/maxwell/

### PR DESCRIPTION
**Pull request by CharlieSYH**
## Info

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/pull/6057

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/335

- **Notes**: 
@tabVersion Is the WITH parameter `properties.group.id` required or optional?
image

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

**Review by tabVersion — APPROVED** [review]
LGTM

**Comment by hengm3467** [comment]
@CharlieSYH Please expand the topic to include definitions of CDC, introduction to Debezium and Maxwell CDC. Example of the Maxwell CDC introduction: https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/table/formats/maxwell/

**Comment by CharlieSYH** [comment]
> @CharlieSYH Please expand the topic to include definitions of CDC, introduction to Debezium and Maxwell CDC. Example of the Maxwell CDC introduction: https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/table/formats/maxwell/

Please review the article again, thanks! @hengm3467

File: docs/sql/create-source/create-source-cdc.md [hunk 1]
```diff
@@ -39,6 +40,11 @@ ROW FORMAT DEBEZIUM_JSON;
 |scan.startup.timestamp_millis	|None	|Int64	|Specify the offset in seconds from a certain point of time.	|False|
 |properties.group.id	|None	|String	|Name of the Kafka consumer group	|True|
 
+### `ROW FORMAT` parameters
+
+- `DEBEZIUM_JSON` — [Debezium](https://debezium.io) is a log-based CDC tool that can capture row changes from various database management systems such as PostgreSQL, MySQL, and SQL Server and generate events with consistent structures. Supported data type: JSON.
```
Comment (hengm3467) [inline docs/sql/create-source/create-source-cdc.md]: It's not accurate to say JSON is a data type. In this context, it is a stream serialization format, so you can say: Supported serialization format: JSON.

File: docs/sql/create-source/create-source-cdc.md [hunk 2]
```diff
@@ -5,12 +5,13 @@ description: Connect RisingWave to a CDC source.
 slug: /create-source-cdc
 ---
 
-   
-Use the SQL statement below to connect RisingWave to a CDC source.
+A change data capture (CDC) tool can record row-level changes (INSERT, UPDATE, and DELETE activities) that apply to tables in a upstream database and stream the data change event records to Kafka topics. You can subscribe to the Kafka topics to receive the data changes.
```
Comment (hengm3467) [inline docs/sql/create-source/create-source-cdc.md]: I'd suggest we add a brief intro to CDC itself as a technology or process before talking about a CDC tool. Something to consider: 
Change Data Capture (CDC) refers to the process of identifying and capturing changes as they are made in a database or source application, then delivering those changes in real-time to a downstream process, system, or data lake. CDC tools and platforms can record ....record to messaging topics such as Kafka topics. You can subscribe these topics to ...

**Review by hengm3467 — APPROVED** [review]
Added two suggestions. Otherwise looks good to me. Thanks!

**Pull request merged**